modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/TSDBIndexingIT.java

```diff
@@ -436,6 +450,117 @@ public void testSkippingShards() throws Exception {
+                new Template(
+                    Settings.builder()
+                        .put("index.mode", "time_series")
+                        .put("index.number_of_replicas", 0)
+                        // Reduce sync interval to speedup this integraton test,
+                        // otherwise by default it will take 30 seconds before minimum retained seqno is updated:
+                        .put("index.soft_deletes.retention_lease.sync_interval", "100ms")
+                        .build(),
+                    new CompressedXContent(MAPPING_TEMPLATE),
+                    null
+                ),
+                null,
+                null,
+                null,
+                null,
+                new ComposableIndexTemplate.DataStreamTemplate(false, false),
+                null
+            )
+        );
+        client().execute(PutComposableIndexTemplateAction.INSTANCE, putTemplateRequest).actionGet();
+
+        // index some data
+        int numBulkRequests = 32;
+        int numDocsPerBulk = 256;
+        String indexName = null;
+        {
+            Instant time = Instant.now();
+            for (int i = 0; i < numBulkRequests; i++) {
+                BulkRequest bulkRequest = new BulkRequest(dataStreamName);
+                for (int j = 0; j < numDocsPerBulk; j++) {
+                    var indexRequest = new IndexRequest(dataStreamName).opType(DocWriteRequest.OpType.CREATE);
+                    indexRequest.source(DOC.replace("$time", formatInstant(time)), XContentType.JSON);
+                    bulkRequest.add(indexRequest);
+                    time = time.plusMillis(1);
+                }
+                var bulkResponse = client().bulk(bulkRequest).actionGet();
+                assertThat(bulkResponse.hasFailures(), is(false));
+                indexName = bulkResponse.getItems()[0].getIndex();
+            }
+            client().admin().indices().refresh(new RefreshRequest(dataStreamName)).actionGet();
+        }
+
+        // Check whether there are multiple segments:
+        var getSegmentsResponse = client().admin().indices().segments(new IndicesSegmentsRequest(dataStreamName)).actionGet();
+        assertThat(
+            getSegmentsResponse.getIndices().get(indexName).getShards().get(0).shards()[0].getSegments(),
+            hasSize(greaterThanOrEqualTo(2))
+        );
+
+        // Pre check whether _id stored field uses diskspace:
+        var diskUsageResponse = client().execute(
+            AnalyzeIndexDiskUsageAction.INSTANCE,
+            new AnalyzeIndexDiskUsageRequest(new String[] { dataStreamName }, AnalyzeIndexDiskUsageRequest.DEFAULT_INDICES_OPTIONS, true)
+        ).actionGet();
+        assertThat(diskUsageResponse.getStats(), aMapWithSize(1));
+        var indexDiskUsage = diskUsageResponse.getStats().get(indexName);
+        var fieldDiskUsage = indexDiskUsage.getFields().get("_id");
+        assertThat(fieldDiskUsage.getStoredFieldBytes(), greaterThanOrEqualTo(1L));
```

Comment:
I wrote `MapMatcher` a while back to make pretty outputs from assertions like this. If you want it, it's there.